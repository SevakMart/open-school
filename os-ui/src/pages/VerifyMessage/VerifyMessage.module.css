.mainContainer {
  display: flex;
  flex-direction: row;
  align-items: none;
  justify-content: space-between;
  padding: 4% 5% 7% 6.25%;
}
.mainContainer h2 {
  display: inline;
  width: 30%;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 800;
  font-size: 3rem;
  line-height: 3.688rem;
  color: #333941;
}
.mainContainer .buttonContainer {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: 35%;
}
.mainContainer .buttonContainer button:nth-of-type(1) {
  background-color: #5e617b;
  color: #ffffff;
  border-radius: 1rem;
  width: none;
  height: 4rem;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 1.375rem;
  padding-bottom: 1.375rem;
  cursor: pointer;
  text-transform: uppercase;
}
.mainContainer .buttonContainer button:nth-of-type(2) {
  background-color: transparent;
  color: #5e617b;
  border-radius: 1rem;
  width: none;
  height: 4rem;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 1.375rem;
  padding-bottom: 1.375rem;
  cursor: pointer;
  border: 1px solid #5e617b;
  text-transform: uppercase;
}/*# sourceMappingURL=VerifyMessage.module.css.map */