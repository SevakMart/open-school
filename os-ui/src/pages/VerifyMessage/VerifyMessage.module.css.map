{"version":3,"sources":["VerifyMessage.module.scss","../../styles/Mixins/FlexMixins.scss","VerifyMessage.module.css","../../styles/Mixins/FontMixin.scss","../../styles/Mixins/ButtonMixin.scss"],"names":[],"mappings":"AAKA;ECJE,aAAA;EACA,mBDImB;ECHnB,iBDGwB;ECFxB,8BDE8B;EAC9B,uBAAA;AEDF;AFEE;EACE,eAAA;EACA,UAAA;EGTF,yBHUqB;EGTrB,kBHSmC;EGRnC,gBHQ2C;EGP3C,eHOgD;EGNhD,qBHMsD;EACpD,cAAA;AEIJ;AFFE;ECbA,aAAA;EACA,mBDaqB;ECZrB,mBDY0B;ECX1B,8BDWkC;EAChC,UAAA;AEOJ;AFNI;EINF,yBJQoB;EIPpB,cJQoB;EIPpB,mBJQoB;EIPpB,WJQoB;EIPpB,YJQoB;EIPpB,kBJQoB;EIPpB,mBJQoB;EIPpB,qBJQoB;EIPpB,wBJQoB;EIPpB,eAAA;EJSI,yBAAA;AEON;AFLI;EIpBF,6BJsBoB;EIrBpB,cJsBoB;EIrBpB,mBJsBoB;EIrBpB,WJsBoB;EIrBpB,YJsBoB;EIrBpB,kBJsBoB;EIrBpB,mBJsBoB;EIrBpB,qBJsBoB;EIrBpB,wBJsBoB;EIrBpB,eAAA;EJuBI,yBAAA;EACA,yBAAA;AEMN","file":"VerifyMessage.module.css"}