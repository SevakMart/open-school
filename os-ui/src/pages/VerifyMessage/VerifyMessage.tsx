import { useState } from 'react';
import SignInForm from '../../component/SignIn/SignInForm';
import styles from '../../component/SignIn/SignIn.module.scss';
import CloseIcon from '../../icons/Close';
import LinkedinIcon1 from '../../icons/Linkedin1';
import EmailIcon1 from '../../icons/Email1';
import ForgotPassword from '../../component/SignIn/ForgotPassword/ForgotPassword';
import { SIGN_UP } from '../../constants/Strings';
import Message from './Message/Message';

const VerifyMessage = ({ handleSignInClicks }:{handleSignInClicks(arg:string):void}) => {
  const {
    mainContainer, formContainer, headerContent, iconContent, alreadyHaveAccount,
  } = styles;

  const [isSignedIn, setIsSignedIn] = useState(false);
  const [forgotPasswordIsSet, setForgotPasswordIsSet] = useState(false);
  const [succesfulSignInMessage, setSuccessfulSignInMessage] = useState('');
  const [isVerifyForLogin, setIsVerifyForLogin] = useState(false);

  const handleSignIn = (message:string) => {
    setIsSignedIn(true);
    setSuccessfulSignInMessage(message);
  };

  const forgotPassword = () => {
    setForgotPasswordIsSet(true);
  };

  const returnToSignInForm = () => {
    setForgotPasswordIsSet(false);
    setIsSignedIn(false);
  };

  const reSend = () => {
    // TO DO
  };

  const goToLogin = () => {
    setIsVerifyForLogin(true);
  };

  return (
    <div className={mainContainer}>
      <div className={formContainer}>
        { !isVerifyForLogin && <Message isVerify={isVerifyForLogin} reSend={reSend} goToLogin={goToLogin} />}
        {isVerifyForLogin && (
        <>
          { forgotPasswordIsSet ? <ForgotPassword returnToSignInForm={returnToSignInForm} />
            : isSignedIn ? <h3 data-testid="successfulSignInMessage">{succesfulSignInMessage}</h3>
              : (
                <>
                  <CloseIcon handleClosing={() => handleSignInClicks('closeButton')} />
                  <div className={headerContent}>
                    <h2>Sign In!</h2>
                    <div className={iconContent}>
                      <button type="button"><LinkedinIcon1 /></button>
                      <button type="button"><EmailIcon1 /></button>
                    </div>
                    <p>Or</p>
                  </div>
                  <SignInForm
                    signInForm="default"
                    handleSignIn={handleSignIn}
                    forgotPasswordHandler={forgotPassword}
                  />
                  <p className={alreadyHaveAccount}>
                    {'Don\'t Have An Account?'}
                    <span><button type="button" onClick={() => handleSignInClicks('signUp')}>{SIGN_UP}</button></span>
                  </p>
                </>
              )}
        </>
        )}
      </div>
    </div>
  );
};
export default VerifyMessage;
