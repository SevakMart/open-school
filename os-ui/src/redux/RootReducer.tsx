import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
