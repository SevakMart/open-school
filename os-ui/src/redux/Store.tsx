import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
