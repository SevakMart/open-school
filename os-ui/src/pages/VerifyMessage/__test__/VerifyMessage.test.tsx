import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../redux/Store';
import VerifyMessage from '../VerifyMessage';

test('Create a snapshot test', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <VerifyMessage />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
