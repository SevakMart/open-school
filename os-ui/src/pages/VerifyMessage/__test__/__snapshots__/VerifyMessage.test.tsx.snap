// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a snapshot test 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="mainContainer"
      >
        <div>
          <h4>
            Go to in your registered email and verify
          </h4>
        </div>
        <div
          class="buttonContainer"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
